sdk/core/azure-core-amqp/inc/azure/core/amqp/internal/connection.hpp

```diff
@@ -166,7 +167,7 @@ namespace Azure { namespace Core { namespace Amqp { namespace _internal {
    */
   class ConnectionEvents {
   protected:
-    ~ConnectionEvents(){};
+    virtual ~ConnectionEvents(){};
```

Comment:
Out of curiosity, why is this needed? Normally, non-virtual protected destructors don't cause problems.

Reply:
Either Clang or Gcc get upset if they encounter a non-virtual destructor in a base class.

There are 3 options to fix it:
1. Add `virtual` to the base class destructor.
2. Mark the derived class as `final`
3. Add a destructor to the derived class (for an events class that isn't a problem because there's no base class state to be torn).

I figured that #1 was the easiest way of making the compiler happy.